File: 28. Find the Index of the First Occurrence in a String/28. Find the Index of the First Occurrence in a String.c
#include <string.h>

int strStr(char* haystack, char* needle) {

    int len = strlen(needle);

    for (int i = 0; haystack[i] != '\0'; i++){
        if (haystack[i] == needle[0]){
            if (strncmp(haystack + i, needle, len) == 0) {
                return i; // Found the needle at index i
            }
        }
    }

    return -1;
}

File: 28. Find the Index of the First Occurrence in a String/28. Find the Index of the First Occurrence in a String.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        for i, c in enumerate(haystack) :
            if c == needle[0]:
                if haystack[i: i + len(needle)] == needle:
                    return i
            else :
                continue
        
        return -1

        